mod/src/main/java/zsawyer/mods/mumblelink/mumble/UpdateData.java

```diff
@@ -233,4 +242,84 @@ private Vec3d getTopVec(Minecraft game) {
 
         return new Vec3d((double) (f2 * f3), (double) f4, (double) (f1 * f3));
     }
+
+    private void handleDimensionYOffset(Minecraft game) {
+        // Don't recalculate this every time, optimally this only needs to be done when a server/world is joined or when
+        // the player enters a different dimension.
+        if (dimensionOffsetUpdateCounter <= 0) {
+            // Todo: Call this as well or only on dimension change
+            calculateDimensionYOffset(game);
+            dimensionOffsetUpdateCounter = DIMENSION_OFFSET_UPDATE_RATE;
+        }
+
+        dimensionOffsetUpdateCounter--;
+
+        if (dimensionYOffset != 0) {
+            // Offset the Y-coordinate based on the dimension
+            fAvatarPosition[2] += dimensionYOffset;
+            fCameraPosition[2] += dimensionYOffset;
+        }
+    }
+
+    private void calculateDimensionYOffset(Minecraft game) {
+        /*
+         * Since the Link API coordinates are stored in floats, and those only have 23 bits of precision, at coordinates
+         * around 2^16 meters, the positional audio precision is already reduced to centimeters and becomes increasingly
+         * after that. At 2^23 a float can not contain more precise location data than meters. This is why a maximum
+         * y-offset around 2^16 (65536) was chosen.
+         * Now there is no way to fit a 256 y-range for each possible dimension id in the range of -65536 to 65536, so
+         * instead the three vanilla dimensions, Nether, Overworld and End get their own reserved "Mumble-space" of 256
+         * meters high and the rest of the dimensions are distributed over the space in between with the realistic chance
+         * of overlapping positional audio spaces, but it is what it is...
+         *
+         * The distribution is as follows:
+         *
+         *    |---------------|----------------------|------------|----------------------|----------|
+         * -(2^16)       -(2^16)+256                 0           256                    2^16     2^16+256
+         *          Nether      Negative ID Mod dims    Overworld   Positive ID Mod dims     End
+         */
+
+        // Coordinates are stored in a float array, which starts to lose precision in the magnitude of meters at 2^23.
+        // To be useful for positional audio use 2^16 (65536) as a maximum offset to only lose precision in the magnitude of centimeters.
+        int maxYOffset = 65536;
+
+        if (game.world == null) {
+            // If we don't know what to do, just use the maximum non-vanilla dimension offset so the "Mumble space"
+            // won't overlap with the Overworld or End.
+            dimensionYOffset = maxYOffset - 256;
+            return;
+        }
+
+        int dimID = game.world.dimension.getType().getId();
+
+        switch (dimID) {
+            case 0:

```

Comment:
Is there maybe an enum that we can use here?
Like 
Dimension.overworld.getId()

Reply:
In MC 1.16 there is an enum that can be used, but this pull request is for MC 1.15

In MC 1.15 there are functions Dimension.isNether and Dimension.isSurfaceWorld however the ID would still be needed to identify the End.
So it is better to leave it this way in my honest opinion.

Reply:
Ok, Now I get it. Please use named constants though -> DIM_ID_NETHER, DIM_ID_END, DIM_ID_OVERWORLD 
(maybe even remove the comments because it becomes self-explanatory)